html/ops/manage_consent_types.php

```diff
@@ -0,0 +1,192 @@
+<?php
+// This file is part of BOINC.
+// http://boinc.berkeley.edu
+// Copyright (C) 2017 University of California
+//
+// BOINC is free software; you can redistribute it and/or modify it
+// under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation,
+// either version 3 of the License, or (at your option) any later version.
+//
+// BOINC is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+// See the GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with BOINC.  If not, see <http://www.gnu.org/licenses/>.
+
+// web interfance for managing consent_type table
+
+require_once('../inc/util_ops.inc');
+
+// This function deletes a row from consent_type table.
+function mct_delete() {
+    $ctid = post_int("consent_type_id");
+    $consent_type = BoincConsentType::lookup("id = $ctid");
+    if ($consent_type) {
+        $myname = $consent_type->shortname;
+        $consent_type->delete_aux("id = $ctid");
+        echo "<h2>Consent Type ${myname} deleted.</h2>";
+    }
+}
+
+// This function adds a row to consent_type table.
+function add_consenttype() {
+    $shortname = BoincDb::escape_string(post_str('add_name'));
+    $description = BoincDb::escape_string(post_str('add_description'));
+
+    if (empty($shortname)) {
+        admin_error_page("The new consent type must contain a short name.</font></p>");
+    }
+    if (empty($description)) {
+        admin_error_page("The new consent type must contain a description.</font></p>");
+    }
+
+    BoincConsentType::insert(
+        "(shortname, description, project_specific) VALUES ('$shortname', '$description', 1)"
```

Comment:
enabled and privacypref are not-null but do not have default values.  Therefore this insert fails.  Either give them default values in the db definition or add them with values in this section of the code.

Reply:
Fixed in 7d7c187